### PR TITLE
Self contained filters

### PR DESCRIPTION
**Pull request by bhollis**
Based on @nev-r's great work to transform search filters into self-contained definition objects. From there we can both organize them logically, but also derive a lot of the information (filter help, suggestions, filters by destinyVersion, etc) that were previously hardcoded in different places.

Check out the tests - I hope they're enough to prove that we're at least generating the right filter suggestions, but this will also need a ton of manual testing.

### CODE REVIEW

**Review by nev-r — APPROVED** [review]
all my spot checks pass (now).
all contexts appear to be working, short ratings which i think we can't test right now?
suggestions seem working and i have some priority changes queued up.
looking forward to editing filters in one place now and programmatically generating filter help. possibly in small per-filter modals somehow

File: src/app/search/search-filters/dupes.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+export const makeDupeID = (item: DimItem) =>
+  (item.classified && `${item.hash}`) ||
+  `${item.name}${item.classType}${item.tier}${item.itemCategoryHashes.join('.')}`;
+
+// so, duplicate detection has gotten complicated in season 8. same items can have different hashes.
+// we use enough values to ensure this item is intended to be the same, as the index for looking up dupes
+export const makeSeasonalDupeID = (item: DimItem) =>
+  (item.classified && `${item.hash}`) ||
+  `${item.name}${item.classType}${item.tier}${
+    item.isDestiny2() ? `${item.collectibleHash}${item.powerCap}` : ''
```
Comment (delphiactual) [inline src/app/search/search-filters/dupes.tsx:25]:
```suggestion
    item.isDestiny2() ? `${item.collectibleHash}${item.powerCap}${item.season}` : ''
```

Comment (delphiactual) [reply src/app/search/search-filters/dupes.tsx:25]: from #5764

Comment (bhollis) [reply src/app/search/search-filters/dupes.tsx]: Thanks!

**Pull request merged**